rest-api-spec/src/main/resources/rest-api-spec/api/ml.get_job_snapshot_upgrade_stats.json

```diff
@@ -0,0 +1,40 @@
+{
+  "ml.get_job_snapshot_upgrade_stats":{
```

Comment:
After discussing more with @sethmlarson we've decided to change this to `ml.get_model_snapshot_upgrade_stats`.

`job_snapshot` used originally here is consistent with `ml.upgrade_job_snapshot`.

But `model_snapshot` is consistent with all the other endpoints that relate to model snapshots.